src/lib.rs

```diff
@@ -983,12 +987,37 @@ fn componentize(
         )
     })?;
 
-    fs::write(path, component).with_context(|| {
+    // To make the write atomic, first write to a temp file and then rename the file
+    let temp_dir = cargo_metadata.target_directory.join("tmp");
+    fs::create_dir_all(&temp_dir)
+        .with_context(|| format!("failed to create directory `{temp_dir}`"))?;
+
+    let mut file = NamedTempFile::new_in(&temp_dir)
+        .with_context(|| format!("failed to create temp file in `{temp_dir}`"))?;
+
+    use std::io::Write;
+    file.write_all(&component).with_context(|| {
         format!(
             "failed to write output component `{path}`",
+            path = file.path().display()
+        )
+    })?;
+
+    file.as_file_mut().sync_all().with_context(|| {
+        format!(
+            "failed to sync output component `{path}`",
+            path = file.path().display()
+        )
+    })?;
```

Comment:
Is there a specific need for a `sync_all` here? Compilers don't usually sync their output to disk, because it can be a slow operation on some filesystems, and it's only significant if there's eg. a power failure.

Reply:
There's no specific need; I'll remove.